### PR TITLE
Community and clubs pages

### PR DESCRIPTION
**Pull request by alexandrawang1**
# [Issue 18](https://github.com/MathSoc/mathsoc-website/issues/18)

Recreate the MathSoc Clubs page as follows:
https://mathsoc.uwaterloo.ca/clubs/

Be sure that the club website links are working. If the link is broken, try finding their website by searching on google "UW "

No additional content needs to be added to this page.

We want the Community Links page to get recreated:
https://mathsoc.uwaterloo.ca/get-involved/

This page is for MathSoc's affiliates. Each image is a link to the associated affiliate's website. This page doesn't need to be redesigned.

Additional Content:
Please add one more Link for MathSoc Cartoons, have their image link to the Cartoons page.

## Notes
TSA does not have a site. Link changed from email to facebook page

Stepper modeled after discord access page

### CODE REVIEW

File: public/assets/sass/pages/get-involved/clubs.scss [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      height: var(--size);
+      border-radius: 50%;
+      background-color: var(--mathsoc-grey-2);
+      pointer-events: all;
+      border: 1px solid;
+      border-color: var(--mathsoc-grey-2);
+      transition: all .25s ease-out;
+    }
+    .stepper_item:hover::before {
+    //   content: "";
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/clubs.scss:37]: could you remove the commented stuff?

File: public/assets/sass/pages/get-involved/clubs.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      bottom: 0;
+    }
+
+    .stepper_content {
+      display: inline-block;
+    }
+
+    .stepper_title {
+      margin: .3rem 0 0 0;
+    //   white-space: nowrap;
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/clubs.scss:64]: here too

Comment (alexandrawang1) [reply public/assets/sass/pages/get-involved/clubs.scss]: Done

File: server/types/schemas.ts [hunk 3]
```diff
@@ -225,6 +225,28 @@ const SharedExecsSchema = z.object({
...
+});
+
+const CommunitySchema = z.object({
+  communityHeader: z.string(),
+  communities: z.array(
+    z.object({
+      title: z.string(),
+      descriptionMarkdown: z.string(),
+      link: z.string(),
+    })
```
Comment (ad-world) [inline server/types/schemas.ts:246]:
```suggestion
    }).strict()
```

File: server/types/schemas.ts [hunk 4]
```diff
@@ -225,6 +225,28 @@ const SharedExecsSchema = z.object({
...
+  )
+});
```
Comment (ad-world) [inline server/types/schemas.ts:248]:
```suggestion
}).strict();
```

File: public/assets/sass/pages/get-involved/clubs.scss [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      background-repeat: no-repeat;
+      background-position: bottom 12px right 12px;
+    }
+    .stepper_item:hover::before {
+      pointer-events: all;
+      background-color: var(--mathsoc-pink-2);
+      transition: all .25s ease-in;
+    }
+    .stepper_item:not(:last-child):after {
+      content: "";
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/clubs.scss:44]: same comment as above

File: public/assets/sass/pages/get-involved/clubs.scss [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    .stepper_item {
+      margin: 0;
+      position: relative;
+      display: flex;
+      gap: 1rem;
+      padding-bottom: 2rem;
+      pointer-events: none;
+    }
+    .stepper_item::before {
+      content: "";
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/clubs.scss:24]: is this line necessary? not sure about your logic here but if its necessary, feel free to resolve

File: public/assets/sass/pages/get-involved/community.scss [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    .stepper_item {
+      margin: 0;
+      position: relative;
+      display: flex;
+      gap: 1rem;
+      padding-bottom: 2rem;
+    }
+
+    .stepper_item::before {
+      content: "";
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/community.scss:25]: same comment

File: public/assets/sass/pages/get-involved/community.scss [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      border: 4px solid #0c2b6e;
+    }
+
+    .stepper_item:hover::before {
+      background-color: var(--mathsoc-pink-2);
+      transition: all .25s ease-in;
+    }
+
+    .stepper_item:not(:last-child):after {
+      content: "";
```
Comment (ad-world) [inline public/assets/sass/pages/get-involved/community.scss:47]: same comment

**Review by ad-world — CHANGES_REQUESTED** [review]
good first pr! just left a few suggestions + clarification questions

File: views/pages/get-involved/clubs.pug [hunk 9]
```diff
@@ -0,0 +1,16 @@
+extends /views/templates/frontend-base.pug
+
+append styles
+    link(href="/assets/css/pages/get-involved/clubs.css" rel="stylesheet")
+
+block body
+    +banner-image("/assets/img/banners/mathsoc-wall.jpeg")
+    div(class="content")
+    section(class="white")
```
Comment (ad-world) [inline views/pages/get-involved/clubs.pug:9]: could you add a header to this page? similar to what you did for the community page

Comment (alexandrawang1) [reply views/pages/get-involved/clubs.pug:9]: Done - header is "Clubs"

File: public/assets/sass/pages/get-involved/clubs.scss (hunk 6 above)
Comment (alexandrawang1) [reply public/assets/sass/pages/get-involved/clubs.scss:24]: yes - its some CSS requirement such that the background isnt visible if there's no content, even if its an empty string. the discord access SCSS uses the same rule

File: server/types/schemas.ts (hunk 4 above)
Comment (alexandrawang1) [reply server/types/schemas.ts]: Done

File: server/types/schemas.ts (hunk 3 above)
Comment (alexandrawang1) [reply server/types/schemas.ts]: Done

**Review by ad-world — APPROVED** [review]
approved!

**Pull request merged**